### PR TITLE
Move AGENT_GUID to non-workspace-specific location to allow for single-machine agent ID

### PR DESCRIPTION
**Pull request by lagalbra**
npmd: check if agentid file exists to get omsagent guid (PR https://github.com/Microsoft/OMS-Agent-for-Linux/pull/508)
If agentid file exists then utilize that for NPM computer id.
To identify an instance of omsagent we utilize agent id and
workspace id.
Ensure agent ID is moved in upgrade from versions > 1.3

Unit and system tests passed
pBuild passed
Verified that agentid file was created under /etc/opt/microsoft/omsagent/ upon clean install
Verified that agentid file was created under /etc/opt/microsoft/omsagent/ using /etc/opt/microsoft/omsagent/conf/omsadmin.conf in --upgrade from version 1.4.0-12
Verified that agentid file was used to set the agent ID for additional multi-homed workspaces
Verified that agentid file remained after --remove

@Microsoft/omsagent-devs

### CODE REVIEW

**Comment by chparimi** [comment]
@lagalbra Now closing PR 508 as the changes have been added here.

File: test/code/plugins/in_npmd_server_plugintest.rb [hunk 1]
```diff
@@ -17,6 +18,7 @@ class NPMDServerTest < Test::Unit::TestCase
 
     TMP_DIR = "tmp_npmd_test"
     FAKE_ADMIN_CONF_LOCATION = "#{TMP_DIR}/omsadmin.conf"
+    # TODO will work with Chaitainya to resolve this
```
Comment (chparimi) [inline test/code/plugins/in_npmd_server_plugintest.rb]: You can remove this comment now.

**Comment by chparimi** [comment]
@lagalbra NPM side changes are fine.

File: test/code/plugins/in_npmd_server_plugintest.rb (hunk 1 above)
Comment (lagalbra) [inline test/code/plugins/in_npmd_server_plugintest.rb]: Fixed.

**Comment by lagalbra** [comment]
Rebased against master

**Comment by lagalbra** [comment]
Ping! Please review.

**Comment by lagalbra** [comment]
@NarineM Can you please give a review?

**Comment by NarineM** [comment]
@lagalbra 
Based on our conversation, this PR should have been for just refactoring agendID serialization such it gets stores in it's own file. Can you explain more how this is impacting computer ID?

**Comment by lagalbra** [comment]
I used "computer ID" merely as a synonym for agent ID.

**Comment by NarineM** [comment]
Can you update PR comments to only use AgentID?

**Comment by lagalbra** [comment]
Rebased against master; will merge once Travis completes.

**Pull request merged**